remoter-server/src/cellSizeDistribution.r

```diff
@@ -45,16 +45,28 @@ generate_default_values_cellSizeDistribution <- function(seurat_obj, config) {
   # returned is both the rank(s) as well as inflection point
   # orig.ident          nCount_RNA  rank
   # SeuratProject       1106        10722
-  tmp <- Tool(seurat_obj_tmp, slot = "CalculateBarcodeInflections")$inflection_points
+  sample_subset <- Tool(seurat_obj_tmp, slot = "CalculateBarcodeInflections")$inflection_points
   # extracting only inflection point(s)
-  return(tmp$nCount_RNA)
+  return(sample_subset$nCount_RNA)
 }
 
 task <- function(seurat_obj, config, task_name, sample_id) {
+    print(paste("Running",task_name,"config: ",sep=" "))
+    print(config)
+    #The format of the sample_id is
+    # sample-WT1
+    # we need to get only the last part, in order to grep the object.
+    tmp_sample <- sub("sample-","",sample_id)
+
     import::here(map2, .from = purrr)
-    minCellSize <- as.numeric(config$filterSettings["minCellSize"])
+    minCellSize <- as.numeric(config$filterSettings$minCellSize)
+
     # extract plotting data of original data to return to plot slot later
-    plot1_data  <- sort(seurat_obj$nCount_RNA, decreasing = TRUE)
+    obj_metadata <- seurat_obj@meta.data
+    barcode_names_this_sample <- rownames(obj_metadata[grep(tmp_sample, rownames(obj_metadata)),]) 
```

Comment:
Maybe checking that this is not empty, if yes, then debug-print tmp_sample and first line of obj_metadata (using testhat)?